lib/connection.js

```diff
@@ -493,6 +493,9 @@ Connection.prototype.transaction = function transaction(fn, options) {
 
           for (const path of state.modifiedPaths) {
             doc.$__.activePaths.paths[path] = 'modify';
+            if (doc.$__.activePaths.states.modify == null) {
+              doc.$__.activePaths.states.modify = {};
+            }
```

Comment:
if i understand this check correctly, then this could be done out of the loop (`for (const path of state.modifiedPaths)`) because it does not need to be repeated, only a condition would need to be added to check the length of `state.modifiedPaths` as to not unnecessary set it when empty

Reply:
Good suggestion :+1: